atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/SweepStrategyManagers.java

```diff
@@ -51,7 +52,17 @@ public static SweepStrategyManager createDefault(KeyValueService kvs) {
                     return cache;
                 });
 
-        return tableRef -> sweepStrategySupplierLoadingCache.get().get(tableRef);
+        return new SweepStrategyManager() {
+            @Override
+            public SweepStrategy get(TableReference tableRef) {
+                return sweepStrategySupplierLoadingCache.get().get(tableRef);
+            }
+
+            @Override
+            public void invalidateCaches(Set<TableReference> tableRefs) {
+                sweepStrategySupplierLoadingCache.get().invalidateAll(tableRefs);
+            }
```

Comment:
```
  /**
   * Discards any cached values for the {@code keys}. The behavior of this operation is undefined
   * for an entry that is being loaded (or reloaded) and is otherwise not present.
   *
   * @param keys the keys whose associated values are to be removed
   * @throws NullPointerException if the specified collection is null or contains a null element
   */
  void invalidateAll(@NonNull Iterable<?> keys);
```

Invalidating something that's not there looks like it could end up either way. It's a bit nasty, but should we take a read-write lock?

Reply:
Discussed offline: The "undefined-ness" here just means that load and invalidate could race and end up either way. This is fine for the internal use case we have planned here.